src/main.cpp

```diff
@@ -3,6 +3,8 @@
 #include "data.hpp"
 #include "pulse.hpp"
 
+#include <libgen.h>
```

Comment:
Better move just above or below `<cstring>`, an implicit section for system headers. `"..."` (double quotes) are local to this project. However, looking at `pulse.cpp` I'm not sure about the desired style.

Reply:
@jbeich I just let the clangd formatter decide where everything goes 😅. But yeah, `libgen` should be at the top

Reply:
done, thanks!